pkg/handlers/default.go

```diff
@@ -49,3 +56,68 @@ func KillContainer(pid int) registry.HandlerFunc {
 		return registry.HandlerResultErrno(unix.EPERM)
 	}
 }
+
+func FreezeContainer(pid int) registry.HandlerFunc {
+	re := regexp.MustCompile(`^(\d*):(freezer)?:(/..*)$`)
+	return func(fd libseccomp.ScmpFd, req *libseccomp.ScmpNotifReq) (result registry.HandlerResult) {
+		cgroupPath := ""
+		if cgroupFile, err := os.Open(filepath.Join("/proc", fmt.Sprintf("%d", pid), "cgroup")); err == nil {
+			defer cgroupFile.Close()
```

Comment:
just curious, are you sure this works? You are defining cgroupFile inside the if, so this var only exists in this scope. Will it be possible to call `cgroupFile.Close()` when the function ends outside of this if? I guess that works? Or panic?

Reply:
The variable defined in the 'if' condition is accessible in the 'if' block, including defer statements if necessary (magic of the Go garbage collector).

Note that direct parameters of a defer function are immediately evaluated, even if the defer function is evaluated later:
https://go.dev/play/p/2jHQzzv9Hk1

In this case, cgroupFile is the receiver of the method Close:
```
func (*File) Close
```
So it receives a pointer to a `File`, but receivers are handled in the same way as arguments:
https://go.dev/play/p/EpH1bXIq54-

Reply:
Cool, thanks for the examples and links! :)